konduit-serving-core/src/main/java/ai/konduit/serving/util/LogUtils.java

```diff
@@ -0,0 +1,146 @@
+/* ******************************************************************************
+ * Copyright (c) 2020 Konduit K.K.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package ai.konduit.serving.util;
+
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.classic.encoder.PatternLayoutEncoder;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.FileAppender;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.input.ReversedLinesFileReader;
+import org.nd4j.shade.guava.base.Strings;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@Slf4j
+public class LogUtils {
+    /**
+     * Gets the file where the logs are.
+     * @return the logs file.
+     */
+    public static File getLogsFile() {
+        Logger rootLogger = (Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
+        FileAppender<ILoggingEvent> fileAppender = (FileAppender<ILoggingEvent>) rootLogger.getAppender("FILE");
+
+        if(fileAppender != null) {
+            return new File(fileAppender.getFile());
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Reads the last n lines from a file
+     * @param file file where the data to be read is.
+     * @param numOfLastLinesToRead the number of last lines to read
+     * @return read lines
+     */
+    public static String readLastLines(File file, int numOfLastLinesToRead) {
+        List<String> result = new ArrayList<>();
+
+        try (ReversedLinesFileReader reader = new ReversedLinesFileReader(file, Charset.defaultCharset())) {
+            String line;
+            while ((line = reader.readLine()) != null && result.size() < numOfLastLinesToRead) {
+                result.add(line);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+            return "";
```

Comment:
Should we return an error message and the stack trace instead of empty string?
Suppose a user is trying to get logs via the UI, and something goes wrong. At present, they'll just get empty string in UI, which could be quite confusing for them.